src/validate.py

```diff
@@ -0,0 +1,157 @@
+"""Commandline tool to check and validate csv files."""
+
+import argparse
+import toml
+import sys
+import pandas as pd
+import pprint
+
+from pathlib import Path
+
+from validation_utils import *
+
+# Location of this file
+PATH = Path.absolute(Path(__file__))
+
+# Files to validate
+CSV_FILE_NAMES = ['environment_events.csv', 'environment.csv', 'host_events.csv', 'hosts.csv']
+
+# Max number of reuslts printed to screen
+MAX_NUM_RES = 25
+
+
+# Colors
+INFO = '\033[93m'
+FAIL = '\033[91m'
+ENDC = '\033[0m'
+
+def main() -> None:
+    """CLI."""
+    parser = argparse.ArgumentParser(
+                prog="CSV file validation",
+                description="Validates the formatting and typing of csv files."
+             )
+    
+    parser.add_argument(
+            "-d",
+            "--data",
+            help="The path to the data folder containing the csv files.",
+            type=Path,
+            required=True
+            )
+
+    parser.add_argument(
+            "-c",
+            "--configfile",
+            help="The path to the validation.toml.",
+            type=Path,
+            default=PATH.parent.parent / "data" / "metadata_test" / "_validation_schema_v2.toml"
+            )
+
+    parser.add_argument(
+            "-s",
+            "--sep",
+            help="Separator between columns in csv files. \n Tab separator: tab. Default: ;",
+            type=str,
+            default=";"
+            )
+
+    parser.add_argument(
+            "-v",
+            "--verbose",
+            help="Get more output in the single steps of the pipeline to debug.",
+            action='store_true'
+            )
+
+    args, _ = parser.parse_known_args() 
+   
+    # For tab separated files
+    if args.sep == "tab":
+        args.sep = "\t"
+
+    # read in toml file
+    validation = read_toml(args.configfile)
+    if args.verbose == True:
+        print(f"DEBUG: validation.toml.")
+        pprint.pprint(validation)
+    # read in data
+    data = read_csv_files(args.data, args.sep)
+    if args.verbose == True:
+        for df_name in data:
+            print(f"DEBUG:shape of {df_name}: {data[df_name].shape}")
+    print("Read in csv files done.")
+
+    # check if all columns exist
+    result = check_column_exists(data, validation)
+    counter = 0
```

Comment:
Perhaps it makes more sense to have a global counter instead of a counter per test?